web/packages/teleport/src/Notifications/Notifications.tsx

```diff
@@ -93,6 +98,33 @@ export function Notifications({ iconSize = 24 }: { iconSize?: number }) {
 
   const ref = useRefClickOutside<HTMLDivElement>({ open, setOpen });
 
+  function onIconClick() {
+    setOpen(!open);
+
+    if (!open && notifications.length) {
```

Comment:
Does `!open` work because react didn't manage to update the state that we change above?
If so, maybe we could revert the order and set the state after sending a request?

Reply:
> Does !open work because react didn't manage to update the state that we change above?
Yes, the state isn't updated immediately by the time we get to the next line.

> If so, maybe we could revert the order and set the state after sending a request?
Could you elaborate? The `setOpen` just opens the pane in the UI, If we set this state after the request then there would be a delay.

I think a better approach which I'll do is use a `useEffect` which runs this logic when `open` changes.

Reply:
Sorry, I made a shortcut. Actually, I was thinking about something like this:
```ts
function onIconClick() {
  if(!open) {
   setOpen(true);
   if (notifications.length) {...}
  } else {
   setOpen(false);
}

I believe this would allow us to avoid `useEffect`.